src/components/ListOfGoods/ListOfGoods.tsx

```diff
@@ -0,0 +1,120 @@
+import { FC, useState } from 'react';
+import 'bulma/css/bulma.css';
+import classNames from 'classnames';
+
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type Props = {
+  goodsFromServer: string[];
+};
+
+export const ListOfGoods: FC<Props> = ({ goodsFromServer }) => {
+  const [isReversed, setReverse] = useState<boolean>(false);
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+
+  const getReorderedGoods = (goods: string[]) => {
+    const visibleGoods = [...goods];
+
+    if (sortType !== SortType.NONE) {
+      visibleGoods.sort((good1, good2) => {
+        switch (sortType) {
+          case SortType.LENGTH:
+            return good1.length - good2.length;
+          case SortType.ALPABET:
+            return good1.localeCompare(good2);
+          default:
+            return 0;
+        }
+      });
+    }
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
+
+  const handleSortAlphabetically = () => {
+    setSortType(SortType.ALPABET);
+  };
+
+  const handleSortLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const handleReset = () => {
+    setSortType(SortType.NONE);
+    setReverse(false);
+  };
+
+  const handleReverse = () => {
+    setReverse(current => (
+      !current));
+  };
+
+  const reorderedGoods = getReorderedGoods(goodsFromServer);
+
+  return (
+    <>
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames(
+            'button',
+            'is-info',
+            { 'is-light': sortType !== SortType.ALPABET },
+          )}
+          onClick={handleSortAlphabetically}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={classNames(
+            'button',
+            'is-success',
+            { 'is-light': sortType !== SortType.LENGTH },
+          )}
+          onClick={handleSortLength}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames(
+            'button',
+            'is-warning',
+            { 'is-light': isReversed !== true },
+          )}
+          onClick={handleReverse}
+        >
+          Reverse
+        </button>
+
+        {(sortType || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={handleReset}
+          >
+            Reset
+          </button>
+        )}
+
+      </div>
+
+      <ul>
+        {reorderedGoods.map(good => (
```

Comment:
Just call getReorderedGoods here

Reply:
I've done this before and I thought it would be better to create a separate variable.